src/server/local-auth-helpers.js

```diff
@@ -0,0 +1,187 @@
+import AuthHasher from 'passport-local-authenticate'
+import { User, Invite, Organization } from './models'
+import { capitalizeWord } from './api/lib/utils'
+
+const errorMessages = {
+  invalidInvite: 'Invalid invite code. Contact your administrator.',
+  invalidCredentials: 'Invalid username or password',
+  emailTaken: 'That email is already taken.',
+  passwordsDontMatch: 'Passwords don\'t match.',
+  invalidResetHash: 'Invalid username or password reset link. Contact your administrator.',
+  noSamePassword: 'Old and new password can\'t be the same'
+}
+
+const validUuid = async (nextUrl, uuidMatch) => {
+  let foundUUID
+  if (nextUrl.includes('join')) {
+    foundUUID = await Organization.filter({ uuid: uuidMatch[0] })
+  } else if (nextUrl.includes('invite')) {
+    foundUUID = await Invite.filter({ hash: uuidMatch[0] })
+  }
+  if (foundUUID.length === 0) {
+    return [null, false, errorMessages.invalidInvite]
+  }
+}
+
+const login = async ({
+  password,
+  existingUser,
+  nextUrl,
+  uuidMatch
+}) => {
+  if (existingUser.length === 0) {
+    return [null, false, errorMessages.invalidCredentials]
+  }
+
+  // Verify UUID validity
+  if (nextUrl) validUuid(nextUrl, uuidMatch)
```

Comment:
When do we validate UUID validity for login?  Do we need it here?
If so, we need an `await` here -- and also need to test whether it succeeded -- and also let's document when we need to validate it here.

Reply:
Good catch. UUID's are validated on login in case it is an invite to an organization or a new campaign.